lib/builder/builder.ex

```diff
@@ -0,0 +1,79 @@
+defmodule ExLinkHeader.Builder do
+
+  alias ExLinkHeader.BuildError
+
+  @moduledoc """
+  HTTP link header builder
+  """
+
+  def build(%ExLinkHeader{} = h) do
+    {_, acc} = Enum.map_reduce(Map.keys(h), [],
+      fn(k, acc) ->
+        entry = Map.get(h, k)
+        case dobuild(entry, k) do
+          v when is_binary(v) -> {v, [v | acc]}
+          _ -> {nil, acc}
+        end
+      end
+    )
+    Enum.join(acc, ", ")
+  end
+
+  defp dobuild(%ExLinkHeaderEntry{host: nil} = _h, _relation) do
+  end
+
+  defp dobuild(%ExLinkHeaderEntry{host: ""} = _h, _relation) do
+  end
+
+  defp dobuild(%ExLinkHeaderEntry{} = h, relation) do
+    q = concatenate(h.q_params, "&", false)
+
+    scheme = case h.scheme do
+      v when is_atom(v) -> Atom.to_string(v)
+      v when is_binary(v) -> v
+      _ -> raise BuildError, "Invalid scheme format/type"
+    end
+    h = Map.put(h, :scheme, scheme)
+
+    url = case h.path do
+      "" -> h.scheme <> "://" <> h.host
+      p when is_binary(p) -> h.scheme <> "://" <> h.host <> "/" <> h.path
+      :nil -> h.scheme <> h.host
+      _ -> h.scheme <> h.host
+    end
+
+    url = case q do
+      "" -> url
+      v when is_binary(v) -> url <> "?" <> v
+      _ -> url
+    end
+
+    attrs = case concatenate(h.t_attributes, "; ", true) do
+      "" -> ""
+      v when is_binary(v) -> "; " <> v
+      _ -> ""
+    end
+
+    "<" <> url <> ">; rel=\"" <> Atom.to_string(relation) <> "\"" <> attrs
+  end
+
+  defp dobuild(_any, _whatever) do
+  end
+
+  defp concatenate(%{} = map , sep, quoted) when is_binary(sep) do
+    Enum.map_join(Map.keys(map), sep, fn(key) ->
+      # TODO: sanitize me (urlencode stuff ?)
```

Comment:
well, not really a TODO, I was just wondering if can happen that some k/v pair may contain elements that must be escaped, as the RFC states. Or the escaping is handled by the underlying http library (like put_resp_header pf Plug.Conn) ? 
Right now, if the underlying lib does not escape the header, the user is responsible to pre-escape them.

But didn't had time to investigate it yet.